tool/ood-gen/lib/event.ml

```diff
@@ -0,0 +1,64 @@
+type location = { lat : float; long : float }
+[@@deriving of_yaml, show { with_path = false }]
+
+type date_with_optional_time = Date of Ptime.date | DateTime of Ptime.t
+
+type metadata = {
+  title : string;
+  url : string;
+  textual_location : string;
+  location : location option;
+  starts : date_with_optional_time;
+  ends : date_with_optional_time option;
```

Comment:
Since there is no obvious representation for `date_with_optional_time` in YAML, these could be `string` and `string option` instead.

Reply:
Okay. Thanks